src/ray/raylet/worker_pool.cc

```diff
@@ -899,14 +934,29 @@ std::shared_ptr<WorkerInterface> WorkerPool::PopWorker(
     if (worker == nullptr) {
       // There are no more non-actor workers available to execute this task.
       // Start a new worker process.
-      proc = StartWorkerProcess(task_spec.GetLanguage(), rpc::WorkerType::WORKER,
-                                task_spec.JobId(), {}, /* dynamic_options */
-                                task_spec.SerializedRuntimeEnv(),
-                                task_spec.OverrideEnvironmentVariables());
+
+      if (task_spec.HasRuntimeEnv()) {
+        // create runtime env.
+        create_runtime_env = true;
+        agent_manager_->CreateRuntimeEnv(
+            task_spec.SerializedRuntimeEnv(),
+            [start_worker_process_fn, &state, task_spec](bool done) {
+              if (!done) {
+                // TODO(guyang.sgy): Reschedule to other nodes when create runtime env
+                // failed.
+                RAY_LOG(ERROR) << "Create runtime env rpc failed. Wait for next time to "
+                                  "retry or reschedule.";
```

Comment:
This log doesn't seem necessary. We've logged about the failure in the agent manager.